ddtrace/contrib/botocore/patch.py

```diff
@@ -467,10 +505,53 @@ def patched_api_call(original_func, instance, args, kwargs):
         span.set_tag(ANALYTICS_SAMPLE_RATE_KEY, config.botocore.get_analytics_sample_rate())
 
         try:
-            result = original_func(*args, **kwargs)
-            _set_response_metadata_tags(span, result)
+            if endpoint_name == "sqs" and operation == "ReceiveMessage" and config._data_streams_enabled:
+                queue_name = get_queue_name(params)
+
+                if "MessageAttributeNames" not in params:
+                    params.update({"MessageAttributeNames": ["_datadog"]})
+                elif "_datadog" not in params["MessageAttributeNames"]:
+                    params.update({"MessageAttributeNames": params["MessageAttributeNames"] + ["_datadog"]})
+
+                result = original_func(*args, **kwargs)
+                _set_response_metadata_tags(span, result)
+
+                try:
+                    if "Messages" in result:
+                        for message in result["Messages"]:
+                            max_attributes_flag = True
+                            if "MessageAttributes" in message:
+                                if "_datadog" in message["MessageAttributes"]:
+                                    if "StringValue" in message["MessageAttributes"]["_datadog"]:
+                                        max_attributes_flag = False
+                                        if PROPAGATION_KEY_BASE_64 in json.loads(
+                                            message["MessageAttributes"]["_datadog"]["StringValue"]
+                                        ):
+                                            pathway = json.loads(
+                                                message["MessageAttributes"]["_datadog"]["StringValue"]
+                                            )[PROPAGATION_KEY_BASE_64]
+                                            ctx = pin.tracer.data_streams_processor.decode_pathway_b64(pathway)
+                                            ctx.set_checkpoint(["direction:in", "topic:" + queue_name, "type:sqs"])
+
+                                        else:
+                                            log.debug(
+                                                "Unable to find data streams context for the message. Data "
```

Comment:
Will this be logged for all botocore users who do not have dsm enabled? If so we should remove it. This could get pretty noisy.

Reply:
so i think it might be confusing for a user that has DSM enabled on the receiving side but then doesn't know why it's not showing up on their DSM page on datadog's app, and can't figure out why that's the case

Reply:
but i'm not a logging expert by any means — will defer to your judgement here! happy to take out the logs if they're too noisy. there might also be a lot of them, since there's a log for each message that doesn't have the DSM context (and the most we can reduce this to is 1 message/1 receive API call, which I can do but might still be expensive)

Reply:
Ahhh can we add this context to the log message. This will make the log messages more actionable.

Reply:
it currently says "Unable to find data streams context for the message. Data streams monitoring was not enabled by the application sending the message at the time of the send."  — did you want something additional?

Reply:
i just updated the other log message to be more actionable as well